cpp/src/io/parquet/reader_impl_helpers.cpp

```diff
@@ -165,9 +165,10 @@ type_id to_type_id(SchemaElement const& schema,
     case FLOAT: return type_id::FLOAT32;
     case DOUBLE: return type_id::FLOAT64;
     case BYTE_ARRAY:
-    case FIXED_LEN_BYTE_ARRAY:
-      // Can be mapped to INT32 (32-bit hash) or STRING
-      return strings_to_categorical ? type_id::INT32 : type_id::STRING;
+      // strings can be mapped to a 32-bit hash
+      if (strings_to_categorical) { return type_id::INT32; }
+      [[fallthrough]];
+    case FIXED_LEN_BYTE_ARRAY: return type_id::STRING;
```

Comment:
does this change in logic impact the new options?

Reply:
It's IMO more correct behavior, since `strings_to_categorical` should only be done on actual string columns, which FLBA cannot be, and it is consistent with other parts of the code (`is_string_col()` and the `ColumnChunkDesc` ctor). I think this change should have been made even without the new options.

Reply:
that's what it looked like, but I wasn't sure. Thanks.